### PR TITLE
Implementation of isPageAccessible getter for refactor

### PR DESCRIPTION
**Pull request by jarmoza**
This PR addresses #261.

The implementation of this boolean getter is a bit more complex than most in our store. `isPageAccessible` is a tweaked version of `pageAccessible` from the previous store. It checks the conditions necessary for access to the given page passed to it.

The tweaks from `pageAccessible` reflect the current state of the implementation of the categorization page (where assessment tools and tool groups are to be reimplemented soon). Currently one should only need at least one category-linked column and for the 'Subject ID' category to be used at least once to progress to the `annotation` page. Secondly, since we do not yet have a good conditional metric for moving to the `download` page, a value `annotationCount` is used. (To move to the categorization page, a non-empty `dataTable` must be present.)

The unit tests implement check for moving to the `categorization` page, `annotation` page, and `download` page in two ways. First, _with_ the sufficient input and then secondly _without_ the sufficient input.

### CODE REVIEW

File: store/index-refactor.js [hunk 1]
```diff
@@ -48,6 +48,58 @@ export const getters = {
...
+                const categorizationStatus = linkCount > 0;
+
+                // 2. Make sure one (and only one) column has been categorized as 'Subject ID'
+                let subjectIDFound = 0;
+                for ( const column in p_state.columnToCategoryMapping ) {
+                    if ( "Subject ID" === p_state.columnToCategoryMapping[column] ) {
+                        subjectIDFound += 1;
+                    }
+                }
+                const singleSubjectIDColumn = ( 1 === subjectIDFound );
```
Comment (surchs) [inline store/index-refactor.js:86]: I think here you could just use the thing from above to make it more succint like so

```javascript
const singleSubjectIDColumn = Object.values(p_state.columnToCategoryMapping).filter(category => "Subject ID" === category).length === 1;
```

To me that would make these two guys together easier to read.

**Review by surchs — APPROVED** [review]
Looks great @jarmoza, thanks! I left two suggestions in the actual getter that I think might be more readable, take a look and see if you agree. But everything works well, so good to go.

File: store/index-refactor.js [hunk 2]
```diff
@@ -48,6 +48,58 @@ export const getters = {
...
+                pageAccessible = p_state.dataTable.length > 0;
+
+                break;
+
+            case "annotation": {
+
+                // 1. Determine if at least one column has been linked to a category
+                const linkCount = Object.values(p_state.columnToCategoryMapping).filter(
+                    category => ( null !== category )).length;
+                const categorizationStatus = linkCount > 0;
```
[the remark this answers is not in the document]
Comment (jarmoza) [reply store/index-refactor.js:77]: Neat! I either didn't know about this function or forgot about it!

[the remark this answers is not in the document]
Comment (surchs) [reply store/index-refactor.js:77]: neither :). But I remember we used `every` somewhere else so I looked on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) for the opposite.

File: store/index-refactor.js (hunk 1 above)
Comment (jarmoza) [reply store/index-refactor.js:86]: Yeah that's just as clear. One trick that I'll apply to this and the categorization status is something D3JS programmers do when chaining functions.

```
const singleSubjectIDColumn = Object.values(p_state.columnToCategoryMapping)
                                                                  .filter(category => "Subject ID" === category)
                                                                  .length === 1; 
```

File: store/index-refactor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarmoza) [reply store/index-refactor.js:77]: One trick D3JS programmers use when chaining functions like this is to align the calls, one underneath the other for clarity.

```
const categorizationStatus = Object.values(p_state.columnToCategoryMapping)
                                                             .some(category => null !== category );
```

(Little bit difficult to demonstrate with non-fixed width font.)

**Pull request merged**